datafusion/core/tests/path_partition.rs

```diff
@@ -204,7 +204,7 @@ async fn csv_filter_with_file_col() -> Result<()> {
     );
 
     let result = ctx
-        .sql("SELECT c1, c2 FROM t WHERE date='2021-10-27' and date!=c1 LIMIT 5")
+        .sql("SELECT c1, c2 FROM t WHERE date='2021-10-27' and c1!='2021-10-27' LIMIT 5")
```

Comment:
I couldn't see a compelling reason why this test needed to test comparison of dictionary columns

Reply:
I think the partitioning columns in a ListingTable are dictionary encoded